docs/CONFIGURATION.md

```diff
@@ -0,0 +1,291 @@
+
+The driver requires IAM permissions to access your Amazon S3 bucket.
+We recommend using [Mountpoint's suggested IAM permission policy](https://github.com/awslabs/mountpoint-s3/blob/main/doc/CONFIGURATION.md#iam-permissions).
+Alternatively, you can use the AWS managed policy AmazonS3FullAccess, available at ARN
+`arn:aws:iam::aws:policy/AmazonS3FullAccess`, but this managed policy grants more permissions than needed for the
+Mountpoint CSI driver. For more details on creating a policy and an IAM role, review
+["Creating an IAM policy"](https://docs.aws.amazon.com/eks/latest/userguide/s3-csi.html#s3-create-iam-policy) and
+["Creating an IAM role"](https://docs.aws.amazon.com/eks/latest/userguide/s3-csi.html#s3-create-iam-role) from the
+EKS User Guide.
+
+The Mountpoint CSI Driver can be configured to ingest credentials via two approaches: globally for the entire 
+Kubernetes cluster, or using credentials assigned to pods.
+
+### Driver-Level Credentials
+
+By setting driver-level credentials, the whole cluster uses the same set of credentials. 
+
+Using this configuration, the credentials that are used are set at installation time, either using Service
+Accounts, or using K8s secrets. 
+
+The CSI Driver uses the following load order for credentials:
+
+1. K8s secrets (not recommended)
+2. Driver-Level IRSA
+3. Instance profiles
+
+
+### Driver-Level Credentials with IRSA
+
+Configuring [IAM Roles for Service Accounts](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) (IRSA)
+is the recommended way to set up the CSI Driver if you want to use Driver-Level credentials.
+
+This approach associates your AWS role to a Service Account used by the CSI Driver. 
+The role is then assumed by the CSI Driver for all volumes with a `driver` authentication source.
+
+```mermaid
+graph LR;
+    CSI[CSI Driver]
+
+    P["`Application Pod
+    *Driver IAM Credentials*`"]
+
+    SA_D[Service Account - Driver]
+    SA_P[Service Account - Pod]
+
+    IAM_D[IAM Credentials - Driver]
+    IAM_P[IAM Credentials - Pod]
+
+    PV[Persistent Volume]
+    PVC[Persistent Volume Claim]
+
+    P --> PVC
+
+    PVC --> PV
+
+    PV --> CSI
+
+    CSI --> SA_D
+    P --> SA_P
+
+    SA_D --> IAM_D
+    SA_P --> IAM_P
+
+    style IAM_D stroke:#0000ff,fill:#ccccff,color:#0000ff
+    style P stroke:#0000ff,fill:#ccccff,color:#0000ff
+```
+
+#### Service Account configuration for EKS Clusters
+
+EKS allows [using Kubernetes service accounts to authenticate requests to S3](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html). 
+To set this up follow these steps:
+
+##### Create a Kubernetes service account for the driver and attach the policy to the service account
+
+> [!NOTE]
+> The same service account name (`s3-csi-driver-sa`) must be specified both in this command and when creating a driver 
+> pod (in the pod spec `deploy/kubernetes/base/node-daemonset.yaml`, Helm value `node.serviceAccount.name`).
```

Comment:
and `--service-account-role-arn` for `eksctl`/`aws eks`

Reply:
This note is just about configuring the name of the service account rather than the ARN role